File: cpp/1528.shuffle_string.cpp
#include <string>
#include <vector>
class Solution {
 public:
    std::string restoreString(const std::string s,
                              const std::vector<int>& indices) {
        std::string result(indices.size(), 'a');
        for (auto i = 0; i < indices.size(); ++i) {
            result[indices[i]] = s[i];
        }
        return result;
    }
};

File: python/1528.shuffle_string.py
from typing import List


class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:

        result = [""] * len(s)

        for i in range(len(s)):
            idx = indices[i]
            letter = s[i]
            result[idx] = letter
        return "".join(result)
